pkg/clusteragent/admission/mutate/config/config.go

```diff
@@ -201,6 +217,56 @@ func injectionMode(pod *corev1.Pod, globalMode string) string {
 	return globalMode
 }
 
+func injectIdentityInContainer(container *corev1.Container, prefix, podStr string) bool {
+	if container == nil {
+		_ = log.Errorf("Cannot inject identity into nil container")
+		return false
+	}
+	// Do not override DD_ENTITY_ID if it's already set
+	if lo.ContainsBy(container.Env, func(v corev1.EnvVar) bool { return v.Name == ddEntityIDEnvVarName }) {
+		log.Debugf("Ignoring container '%s' in pod %s: env var '%s' already exist", container.Name, podStr, ddEntityIDEnvVarName)
+		return false
+	}
+
+	// We can and should override DD_INTERNAL_* variables if they are already set
+	container.Env = lo.Filter(container.Env, func(v corev1.EnvVar, _ int) bool { return v.Name != podUIDEnvVarName })
+	addedEnv := []corev1.EnvVar{
+		// DD_INTERNAL_POD_UID must precede DD_ENTITY_ID to be referenced in the latter.
+		// See https://kubernetes.io/docs/tasks/inject-data-application/define-interdependent-environment-variables/
+		{
+			Name: podUIDEnvVarName,
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.uid",
+				},
+			},
+		},
+		{
+			Name:  ddEntityIDEnvVarName,
+			Value: fmt.Sprintf("en-%s$(%s)/%s", prefix, podUIDEnvVarName, container.Name),
+		},
+	}
+
+	// prepend rather than append so that our new vars precede container vars in the final list, so that they
+	// can be referenced in other env vars downstream.  (see:  Kubernetes dependent environment variables.)
+	container.Env = append(addedEnv, container.Env...)
+	return true
+}
+
+// injectFullIdentity injects `DD_INTERNAL_CONTAINER_NAME`, `DD_INTERNAL_POD_UID` and `DD_ENTITY_ID`
+// as `en-(init.)$(DD_INTERNAL_POD_UID)/$(DD_INTERNAL_CONTAINER_NAME)`.
+func injectFullIdentity(pod *corev1.Pod) bool {
+	injected := false
+	podStr := common.PodString(pod)
+	for i := range pod.Spec.Containers {
+		injected = injectIdentityInContainer(&pod.Spec.Containers[i], "", podStr)
```

Comment:
Should this be `injected = injectIdentityInContainer(&pod.Spec.Containers[i], "", podStr) || injected`?

Reply:
Good catch 👍